### ISSUE TITLE
 Does not work with React Native Modal

### ISSUE THREAD

**Issue by cassiossantos** [computerjazz/react-native-draggable-flatlist#240]
**Description**
It does not work inside the React Native Modal component.

**How To Reproduce**
Create a modal and add the Flatlist as children.

**Tested in physical device**
- [x] Android

**Comment by creambyemute** [comment]
@cassiossantos 

I'm actually using it inside a Modal and it does work for me:

```
 <Modal
            visible={props.rootStore.createInspectionWizardModalStore.isModalVisible}
            onDismiss={onDismiss}
            contentContainerStyle={styles.root}
            dismissable={true}
        >
            <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
                <View style={{ flex: 1 }}>{renderContent()}</View>
            </TouchableWithoutFeedback>
        </Modal>
```

renderContent renders a TabView where the second tab has a DraggableFlatlist inside. Nothing special there, just styles with flex: 1 and some paddings/margins.

**Comment by cassiossantos** [comment]
###  Works

```
import React, {Component} from 'react';
import {
  View,
  TouchableOpacity,
  Text,
  Modal,
  TouchableWithoutFeedback,
} from 'react-native';
import DraggableFlatList from 'react-native-draggable-flatlist';
import Splash from 'react-native-splash-screen';

const exampleData = [...Array(20)].map((d, index) => ({
  key: `item-${index}`, // For example only -- don't use index as your key!
  label: index,
  backgroundColor: `rgb(${Math.floor(Math.random() * 255)}, ${index *
    5}, ${132})`,
}));

class Example extends Component {
  state = {
    data: exampleData,
  };

  componentDidMount() {
    Splash.hide();
  }

  renderItem = ({item, index, drag, isActive}) => {
    return (
      <TouchableOpacity
        style={{
          height: 100,
          backgroundColor: isActive ? 'blue' : item.backgroundColor,
          alignItems: 'center',
          justifyContent: 'center',
        }}
        onLongPress={drag}>
        <Text
          style={{
            fontWeight: 'bold',
            color: 'white',
            fontSize: 32,
          }}>
          {item.label}
        </Text>
      </TouchableOpacity>
    );
  };

  render() {
    return (
      // <Modal visible={true} style={{flex: 1}}>
      <TouchableWithoutFeedback>
        <View style={{flex: 1}}>
          <DraggableFlatList
            data={this.state.data}
            renderItem={this.renderItem}
            keyExtractor={(item, index) => `draggable-item-${item.key}`}
            onDragEnd={({data}) => this.setState({data})}
          />
        </View>
      </TouchableWithoutFeedback>
      // </Modal>
    );
  }
}

export default Example;


```

###  Doesn't work

```

import React, {Component} from 'react';
import {
  View,
  TouchableOpacity,
  Text,
  Modal,
  TouchableWithoutFeedback,
} from 'react-native';
import DraggableFlatList from 'react-native-draggable-flatlist';
import Splash from 'react-native-splash-screen';

const exampleData = [...Array(20)].map((d, index) => ({
  key: `item-${index}`, // For example only -- don't use index as your key!
  label: index,
  backgroundColor: `rgb(${Math.floor(Math.random() * 255)}, ${index *
    5}, ${132})`,
}));

class Example extends Component {
  state = {
    data: exampleData,
  };

  componentDidMount() {
    Splash.hide();
  }

  renderItem = ({item, index, drag, isActive}) => {
    return (
      <TouchableOpacity
        style={{
          height: 100,
          backgroundColor: isActive ? 'blue' : item.backgroundColor,
          alignItems: 'center',
          justifyContent: 'center',
        }}
        onLongPress={drag}>
        <Text
          style={{
            fontWeight: 'bold',
            color: 'white',
            fontSize: 32,
          }}>
          {item.label}
        </Text>
      </TouchableOpacity>
    );
  };

  render() {
    return (
      <Modal visible={true} style={{flex: 1}}>
        <TouchableWithoutFeedback>
          <View style={{flex: 1}}>
            <DraggableFlatList
              data={this.state.data}
              renderItem={this.renderItem}
              keyExtractor={(item, index) => `draggable-item-${item.key}`}
              onDragEnd={({data}) => this.setState({data})}
            />
          </View>
        </TouchableWithoutFeedback>
      </Modal>
    );
  }
}

export default Example;


```

**Comment by cassiossantos** [comment]
@creambyemute Thanks for the reply. Is this Modal the same as React Native or are you using another library?

**Comment by creambyemute** [comment]
@cassiossantos Dang you are right about the library. I'm using the Modal from React-Native-Paper UI-Library. It seems to work inside there.

**Comment by cassiossantos** [comment]
@creambyemute 

Yes, I'm using the latest versions. I researched a little more and I believe it is related to RNGH, apparently it does not work within Modal. There is an open issue on this https://github.com/software-mansion/react-native-gesture-handler/issues/139

I tried the solution proposed by RNGH, but without success. Unfortunately I'll have to find another library, I thought of creating a Modal using a View but then I'll have to change the entire structure.

**Comment by creambyemute** [comment]
@cassiossantos Have you thought of using a different modal like react-native-modal or react-native-paper and it's modal? RN-Paper should be able to be tree-shaken (and thus throw away what you don't need/use) afaik.

**Comment by cassiossantos** [comment]
@creambyemute Sorry for the delay in replying and thanks for your reply, but the libraries you listed would not work for me.

But I found the problem, it really is related to RNGH it is necessary to wrap this component with "GestureHandlerRootView" as in the example below , this solves and the gestures will certainly work.

````javascript

import { GestureHandlerRootView } from 'react-native-gesture-handler';

export default MyComponent() {
  return (
    <Modal>
        <GestureHandlerRootView>
            {/* gestures content  */}
        </GestureHandlerRootView>
    </Modal>
  )
}

````
In my case, the above solution didn't work because I'm using React Navigation also whose uses Root View as well. I was able to solve it by creating a modal with React Navigation and making an overlay of screens, I didn't use the React Native Modal anymore.

@computerjazz  This is an RNGH problem so I believe that it would be good to solve this already, as recommended here  https://docs.swmansion.com/react-native-gesture-handler/docs/#for-library-authors
